src/DynamoCoreWpf/ViewModels/Core/StateMachine.cs

```diff
@@ -174,8 +174,17 @@ internal void BeginConnection(Guid nodeId, int portIndex, PortType portType)
         {
             bool isInPort = portType == PortType.Input;
 
-            if (!(Model.GetModelInternal(nodeId) is NodeModel node)) return;
-            PortModel portModel = isInPort ? node.InPorts[portIndex] : node.OutPorts[portIndex];
+            if (Model.GetModelInternal(nodeId) is not NodeModel node) return;
+            PortModel portModel;
+            try
+            {
+                portModel = isInPort ? node.InPorts[portIndex] : node.OutPorts[portIndex];
+            }
+            catch(Exception ex)
+            {
+                System.Diagnostics.Debug.WriteLine("Failed to make connection: " + ex.Message);
```

Comment:
Where does this log go to?

Reply:
This logs probably to the log file, I used the same output as used a few lines below.
Would you like it to be logged to console or analytics instead?

Reply:
I think it only goes to Visual Studio Output Console while debugging, see https://stackoverflow.com/questions/1159755/where-does-system-diagnostics-debug-write-output-appear. The below example may not be the perfect one to follow. I would prioritize Dynamo logger, if not available, analytics exception logging.

Reply:
Got it, made the changes to log to dynamo console for both places.